homeassistant/components/somfy/api.py

```diff
@@ -0,0 +1,55 @@
+"""API for Somfy bound to HASS OAuth."""
+from asyncio import run_coroutine_threadsafe
+from functools import partial
+
+import requests
+from pymfy.api import somfy_api
+
+from homeassistant import core, config_entries
+from homeassistant.helpers import config_entry_oauth2_flow
+
+
+class ConfigEntrySomfyApi(somfy_api.AbstractSomfyApi):
```

Comment:
Will we have to overwrite the integration library api for each library?

Can we support libraries that handle token refresh already?

Reply:
Note that we only provide our own auth part. The rest of the lib is still used as-is.

Libs that are based on the requests-oauthlib generally allow us to pass in the token and token_updated callback, in that case we don't need to create our own but can create a generic one.

The challenge is with libs that tightly couple auth and request making. Those libs will need to be redone to make it easier to work with Home Assistant. I will write documentation for this.